### PR TITLE
Widget export

### PR DESCRIPTION
**Pull request by Juliana-Dimeglio**
Export of the widgets and an example of their implementation 

https://user-images.githubusercontent.com/85245831/127925742-0c665364-b144-4cd8-9135-9b4fee6b0262.mp4

### CODE REVIEW

File: example/lib/pages/movie_image_page.dart [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+          icon: Icon(
+            Icons.arrow_back,
+            color: Colors.black,
+          ),
+        ),
+      ),
+      body: MovieImage(
+        posterPath: StringConstants.posterPath,
+      ),
+    );
```
Comment (FranciscoGaraventa) [inline example/lib/pages/movie_image_page.dart]: You're using the same code to create all the pages but the only change that you do is the body. So you can create only a  widget of Page (line 10 to 20 is the same) but you need to add a parameter that is going to be a Widget

File: example/lib/src/home_page.dart [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: ListView.builder(
+        scrollDirection: Axis.vertical,
+        shrinkWrap: true,
+        itemCount: _routeList.length,
+        itemBuilder: (BuildContext context, int index) {
+          return Padding(
+            padding: EdgeInsets.all(30),
```
Comment (FranciscoGaraventa) [inline example/lib/src/home_page.dart]: You can add this int to a constant file

File: lib/utils/numeric_constants_widget.dart [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+  static const double fontSizeInfoDetail = 25;
+  static const double borderRadiusTitle = 10;
+  static const double paddingTitleDetail = 8.0;
+  static const double fontSizeTitleDetail = 30;
+  static const double marginPoster = 10;
+  static const double heightPosterDetail = 400;
+  static const double paddingOverviewVertical = 10;
+  static const double paddingOverviewHorizontal = 15;
+  static const double fontSizeOverview = 23;
+}
```
Comment (FranciscoGaraventa) [inline lib/utils/numeric_constants_widget.dart]: You missed a new line at end of file

File: lib/utils/string_constants_widget.dart [hunk 4]
```diff
@@ -0,0 +1,5 @@
+abstract class StringConstantsWidget{
+  static const String defaultImage = "assets/movie_poster.png";
+  static const String errorImage = "assets/error_image.png";
+  static const String fontFamilyTitleDetail = "cursive";
+}
```
Comment (FranciscoGaraventa) [inline lib/utils/string_constants_widget.dart]: You missed a new line at end of file

File: example/lib/pages/movie_image_page.dart (hunk 1 above)
Comment (Juliana-Dimeglio) [reply example/lib/pages/movie_image_page.dart]: done

File: lib/utils/numeric_constants_widget.dart (hunk 3 above)
Comment (Juliana-Dimeglio) [reply lib/utils/numeric_constants_widget.dart]: done

File: lib/utils/string_constants_widget.dart (hunk 4 above)
Comment (Juliana-Dimeglio) [reply lib/utils/string_constants_widget.dart]: done

File: example/lib/src/home_page.dart (hunk 2 above)
Comment (Juliana-Dimeglio) [reply example/lib/src/home_page.dart]: done

File: example/lib/src/home_page.dart [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: ListView.builder(
+        scrollDirection: Axis.vertical,
+        shrinkWrap: true,
+        itemCount: _routeList.length,
+        itemBuilder: (BuildContext context, int index) {
+          return Padding(
+            padding: EdgeInsets.all(NumericConstants.paddingButton),
```
Comment (FranciscoGaraventa) [inline example/lib/src/home_page.dart:58]: You missed a trailing comma here

File: example/lib/src/home_page.dart [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+            child: ElevatedButton(
+              onPressed: () {
+                Navigator.of(context).pushNamed(_routeList[index].route);
```
Comment (FranciscoGaraventa) [inline example/lib/src/home_page.dart:61]: You missed a trailing comma here

File: example/lib/src/home_page.dart [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+              },
+              child: Text(_routeList[index].widgetName),
```
Comment (FranciscoGaraventa) [inline example/lib/src/home_page.dart:63]: You missed a trailing comma here

File: example/lib/pages/basic_page.dart [hunk 8]
```diff
@@ -0,0 +1,27 @@
+import 'package:flutter/material.dart';
+
+class BasicPage extends StatelessWidget {
+   final Widget widget;
+
+  const BasicPage({
+    Key? key,
+    required  this.widget,
```
Comment (FranciscoGaraventa) [inline example/lib/pages/basic_page.dart:8]: Instead to call this "widget", you can change with a more descriptive variable name like bodyWidget, so when you are going to create a BasicPage you can see something like this: BodyPage( bodyWidget: ... ), it's tell you where you are going to use the Widget that you pass as parameter.

**Review by FranciscoGaraventa — APPROVED** [review]
LG

File: example/lib/pages/basic_page.dart (hunk 8 above)
Comment (Juliana-Dimeglio) [reply example/lib/pages/basic_page.dart:8]: done

File: example/lib/src/home_page.dart (hunk 7 above)
Comment (Juliana-Dimeglio) [reply example/lib/src/home_page.dart:63]: done

File: example/lib/src/home_page.dart (hunk 6 above)
Comment (Juliana-Dimeglio) [reply example/lib/src/home_page.dart:61]: done

File: example/lib/src/home_page.dart (hunk 5 above)
Comment (Juliana-Dimeglio) [reply example/lib/src/home_page.dart:58]: done

**Review by MarcosCGlobant — APPROVED** [review]
LGTM

**Pull request merged**